### DISCUSSION TITLE
Can I use gumtree to find what changed in two packages?

### DISCUSSION THREAD

**Discussion by wangsiqidahaoren** [GumTreeDiff/gumtree#352] [Q&A] [answered]
Can I use gumtree to find what changed in two packages? I use the 'gumtree cluster 1.py 2.py' to find two pythons diff. How can I find the change between two version of a project?
I also want to know where I can extract the diff directly, from the source code, where I can extract the data flow of the changed nodes' cluster?

**Comment by wangsiqidahaoren** [comment]
@jrfaller

**Reply by jrfaller** [reply]
Hi! The command to diff two folders is webdiff. However, there is no data flow analysis in GumTree. Cheers!

**Reply by wangsiqidahaoren** [reply]
I'm sorry, I must have made some mistake in my words. I mean, I want to catch the data(the diff, maybe array or structures) before they are shown on the command window or something. Which API should I analyse in the Gumtree's source code

**Reply by jrfaller** [reply, marked as answer]
Ok! In the wiki there is a tutorial on using Gumtree via the Java API. Cheers!
